### DISCUSSION TITLE
page.on_close

### DISCUSSION THREAD

**Discussion by nenocalmeida** [flet-dev/flet#4533] [Q&A]
My vscode is not capturing the on_close event in python
Flet version: 0.25.1
python version: 3.12.5
Operating system: Windows 10

import flet as ft

def main(page: ft.Page):
    page.title = "Flet App"

    page.add(ft.Text("Hello, World!"))

    def window_event(e):
        if e.data == 'close':
            print('fechando a janela')

    page.on_close = window_event    
    
    page.update()

ft.app(main)

**Comment by ndonkoHenri** [comment]
Inspire yourself from this example from the docs: https://flet.dev/docs/reference/types/window/#destroy

**Reply by OmarBelfeki** [reply]
app crush (windows) when clicking yes and the top is app not responding

https://github.com/user-attachments/assets/d4bd721a-4d04-4dc9-905a-d7cb368a953a

**Reply by ndonkoHenri** [reply]
That's pretty strange. Does this happen with Flet 0.25.2: `pip install 'flet[all]' --upgrade`

**Reply by alessioprini** [reply]
Hi, I identified the same behavior. Is it a confirmed bug? I cannot find issues on that. 
flet 0.27.6
python 3.13.3

**Reply by ndonkoHenri** [reply]
@alessioprini feel free to open an issue with the details.

**Reply by FeodorFitsner** [reply]
page.on_close is for web only: https://flet.dev/docs/controls/page/#on_close

**Reply by hieubnt235** [reply]
> page.on_close is for web only: https://flet.dev/docs/controls/page/#on_close

Hi, i'm using on_close handler with selfhost fastapi app (export_asgi_app option). I limit the session timeout to 5 seconds, but after session is deleted, the on_close event aren't called. I just want to catch the event that session is expired but cannot, please give me suggestions, thank you.
flet[all]==0.28.3
python>=3.12
